### PR TITLE
Support FASTA headers without whitespace

### PR DESCRIPTION
**Pull request by kkchau**
Allow support for indexing FASTA files with identifiers that don't contain whitespace while `full_name=False`.

Index creation iterates through the header line until a whitespace character is found, but if the iteration doesn't encounter a whitespace character, the string is terminated at a position one-too-far. If this happens, just reset the index back to the original length.

Also bumps zlib to 1.3

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,9 +17,9 @@ def prepare_zlib():
     global include_dirs
     global sources
 
-    zlib_dir = os.path.join(root_dir, "zlib-1.2.13")
-    zlib_file = os.path.join(root_dir, "zlib-1.2.13.zip")
-    url = "https://github.com/madler/zlib/releases/download/v1.2.13/zlib1213.zip"
+    zlib_dir = os.path.join(root_dir, "zlib-1.3")
+    zlib_file = os.path.join(root_dir, "zlib-1.3.zip")
+    url = "https://zlib.net/zlib13.zip"
```
Comment (DavidCain) [inline setup.py:22]: @aroon-color - can/should we just make a separate PR for this?

Comment (kkchau) [reply setup.py:22]: Good point, we should probably have a set of stacked PRs since we pin the commit associated with this PR

**Comment by kkchau** [comment]
@lmdu wondering if you could take a look at this PR?

**Comment by kkchau** [comment]
Addressed by https://github.com/lmdu/pyfastx/commit/465bac0677d78b76d5a350e2f255d7b7d3d9264b, closing
